proposals/solid-notifications.md

```diff
@@ -0,0 +1,270 @@
+
+#### Request
+
+```http
+POST /gateway
+Content-Type: application/ld+json
+
+{
+    "@context": ["https://www.w3.org/ns/solid/notification/v1"],
+    "type": ["WebSocketSubscription2021", "WebHookSubscription2021"],
+    "features": ["state", "rate"]
+}
+```
+
+#### Response
+
+```http
+HTTP/2
+Content-Type: application/ld+json
+
+{
+    "@context": ["https://www.w3.org/ns/solid/notification/v1"],
+    "type": "WebSocketSubscription2021",
+    "endpoint": "https://websocket.example/subscription",
+    "features": ["state", "rate", "expiration", "feature1", "feature2"]
+}
+```
+
+**Note**: the features array may not be needed as part of this negotiation.
+
+### Notification Subscription
+
+The details of subscribing to a particular endpoint depend on the technology used, but all will follow a similar pattern. For WebSockets, this involves sending an authenticated subscription request to the endpoint retrieved from the gateway service.
+
+The client sends a JSON-LD payload to this endpoint via POST. The only required fields in this interaction are type and topic. The type field MUST contain the type of subscription being requested. The topic field MUST contain the URL of the resource a client wishes to subscribe to changes.
+
+All other fields are defined by the particular subscription type. Some common features are described in the Features section of this document.
+
+An example POST request using a DPoP bound access token is below:
+
+```http
+POST /subscription
+Authorization: DPoP <token>
+DPoP: <proof>
+Content-Type: application/ld+json
+
+{
+    "@context": ["https://www.w3.org/ns/solid/notification/v1"],
+    "type": "WebSocketSubscription2021",
+    "topic": "https://pod.example/resource",
+    "state": "opaque-state",
+    "expiration": "2021-09-21T12:37:15Z",
+    "rate": "PT10s"
+}
+```
+
+A successful response will contain a URL that can be used directly with a JavaScript client
+
+```http
+HTTP/2
+Content-Type: application/ld+json
+
+{
+    "@context": "https://www.w3.org/ns/solid/notification/v1",
+    "type": "WebSocketSubscription2021",
+    "endpoint": "wss://websocket.example/?auth=Ys3KiUq"
+}
+```
+
+In JavaScript, a client can use the data in the response to establish a connection to the WebSocket endpoint.
+
+```javascript
+const ws = new WebSocket(endpoint, type)
+```
+
+A client will define how to handle events, especially the onclose and onmessage events.
+
+```javascript
+ws.onclose(evt => reconnect(evt));
+ws.onmessage(evt => console.log("Message received: ", evt))
+```
+
+## Authentication and Authorization
+
+The Subscription API requires authorization. This document does not define the specific technology used to authorize requests; rather, it defers to the Solid Protocol sections on Authentication and Authorization. It is out of scope for this document to describe how a client fetches an authorization token. Solid-OIDC is one example of an authentication mechanism that could be used with Solid Notifications.
```

Comment:
The solid spec must define the whole protocol, including how the auth plugs into the notification.   Propose that this document *does* define that.   Otherwise we need another parallel document alongside this.  The spec must completely define exactly ow to build a solid server.

Reply:
Noted. Authorization is a topic that will need quite a bit of discussion in the panel and deserves its own dedicated thread. Does this concern block merging this PR in its present form?

Reply:
#13